py-polars/polars/lazyframe/frame.py

```diff
@@ -2581,6 +2583,10 @@ def _set_sink_optimizations(
             new_streaming=False,
         )
 
+    @deprecate_function(
+        "This will shortly become a private function; use is strongly discouraged",
```

Comment:
I'd suggest taking this opportunity to inform the user what to use instead.

```suggestion
        "`LazyFrame.fetch` is deprecated. Use `LazyFrame.collect` instead.",
```

Reply:
Tweaked accordingly, along with a suggestion to use in conjunction with `head`, since the purpose is to return a small subset of rows 👌